proc/breakpoints.go

```diff
@@ -139,6 +139,11 @@ func (dbp *Process) setBreakpoint(tid int, addr uint64, temp bool) (*Breakpoint,
 	return dbp.Breakpoints[addr], nil
 }
 
+func (dbp *Process) writeSoftwareBreakpoint(thread *Thread, addr uint64) error {
```

Comment:
I will do that if you insist but they are not symmetric, setHardwareBreakpoint is a complete operation, to set a software breakpoint we have to read the original instructions and then write the breakpoint and writeSoftwareBreakpoint only does the second part.